pyproject.toml

```diff
@@ -0,0 +1,199 @@
+
+
+# https://pdm-project.org/latest/
+[tool.pdm.version]
+source = "scm"
+
+[tool.pdm.build]
+source-includes = ["scripts/", "tests/"]
+
+#[tool.pdm.scripts.lint]
+#help = "Lint code"
+#cmd = "pre-commit run --all-files"
+
+[tool.pdm.scripts.types]
+help = "Check type hints"
+cmd = "mypy --install-types --non-interactive --config-file=pyproject.toml {args:src tests}"
+
+[tool.pdm.scripts.docs]
+help = "Build and test documentation"
+composite = [
+  "sphinx-build -W --keep-going --color -b html docs docs/_build",
+  "sphinx-build -W --keep-going --color -b linkcheck docs docs/_build",
+#  "sphinx-build -W --keep-going --color -b doctest docs docs/_build",
+]
+
+[tool.pdm.scripts.test-nocov]
+help = "Run tests without coverage"
+cmd = "pytest {args:-n auto}"
+
+[tool.pdm.scripts.test]
+help = "Run tests with coverage"
+cmd = "pytest --cov --cov-report=term-missing --cov-config=pyproject.toml {args:-n auto}"
+
+
+# https://coverage.readthedocs.io/en/stable/
+[tool.coverage.run]
+source_pkgs = ["pymediainfo", "tests"]
+branch = true
+parallel = true
+omit = [
+  "src/pymediainfo/_version.py",
+]
+
+[tool.coverage.paths]
+pymediainfo = ["src/pymediainfo", "*/pymediainfo/src/pymediainfo"]
+tests = ["tests", "*/pymediainfo/tests"]
+
+[tool.coverage.report]
+exclude_also = [
+  "# pragma: no.cover",
+  "# pragma: no.branch",
+  "if __name__ == .__main__.:",
+  "if TYPE_CHECKING:",
+]
+
+
+# https://docs.astral.sh/ruff/
+[tool.ruff]
+line-length = 100
+src = ["src", "tests"]
+exclude = [
+    "tests/data/*",
+    "_version.py",
+]
+
+[tool.ruff.lint]
+select = ["ALL"]
+ignore = [
+    "ISC001",  # conflict with `ruff format`
+    "D203",    # pydocstyle
+    "D105",
+    "D107",
+    "N802",    # pep8-naming
+    "N803",
+    "N806",
+    "ANN101",  # flake8-annotations
+    "ANN102",
+    "ANN401",
+    "ERA001",  # eradicate
+    "G004",    # flake8-logging-format
+    "TD",      # flake8-todos
+    "COM812",  # flake8-commas
+    "PTH123",  # flake8-use-pathlib
+    "SIM103",  # flake8-simplify
+    "PLR0913", # pylint
+    "PLR2004",
+    "PLR0912", # refactor
+    "PLR0915",
+    "PERF203", # Perflint
+    "PD901",   # pandas-vet
+    ### to be removed
+    "C901",    # mccabe
+]
+
+[tool.ruff.lint.per-file-ignores]
+"__init__.py" = ["E402", "D104"]
+"**/tests/*" = [
+    "D",
+    "S",
+    "RUF012",
+    "N999",
+    "E402",
+    "INP001",
+    "PTH",
+    "T201",
+    "ANN",
+    "SLF",
+    "FBT",
+]
+"**/scripts/*" = ["INP001"]
+"**/docs/*" = [
+    "A001",
+    "D100",
+    "INP001",
+]
+
+[tool.ruff.lint.pydocstyle]
+convention = "pep257"
+
+# https://docs.astral.sh/ruff/formatter/
+[tool.ruff.format]
+docstring-code-format = true
+
+
+# https://mypy.readthedocs.io/en/stable/config_file.html
+[tool.mypy]
+# global-only flags
+pretty = true
+show_error_codes = true
+
+[[tool.mypy.overrides]]
+module = ["pymediainfo.*"]
+strict = true
+
+
+# https://github.com/crate-ci/typos/blob/master/docs/reference.md
+[tool.typos.default]
```

Comment:
I don't see typos, ruff or coverage called in appveyor, can you add it there and in tox.ini (if it's supposed to pass)?

Reply:
Ah yes, that was for a next PR and I forgot to remove the sections from `pyproject.toml`.
I made bigger changes and I split them in smaller PRs so it's easier to review.

Anyway these are just configuration sections, if you don't use the associated tools they are just ignored.

Reply:
Of course, it just seemed strange to add them and not use them. Let's remove them from this PR to make things more atomic if you're OK (I can do that if you'd like me to, I don't mind).

Reply:
Sure, please go ahead